src/dlgConnectionProfiles.cpp

```diff
@@ -1016,63 +1047,108 @@ void dlgConnectionProfiles::fillout_form()
+            // so we can invoked it from the command-line with the "-p"
+            // option for tests
+            pItem->setData(csmHiddenRole, true);
+        } else {
+            if ((deletedDefaultMuds.contains(game))
+                ||(!showAllPredefinedMuds && !onlyShownPredefinedProfiles.contains(game, Qt::CaseInsensitive))) {
+                // We do not want this profile to show but we still need to
+                // keep track of it:
+                pItem->setData(csmHiddenRole, true);
+            } else {
+                pItem->setData(csmHiddenRole, false);
             }
+            setupMudProfile(pItem, game, (*details).description, (*details).icon);
+            reduceFontSize(pItem);
+            nameToItemMap.insert(pItem->data(csmNameRole).toString(), pItem);
         }
+    }
 
-#if defined(QT_DEBUG)
-        const QString mudServer = qsl("Mudlet self-test");
-        if (!deletedDefaultMuds.contains(mudServer) && !mProfileList.contains(mudServer)) {
-            mProfileList.append(mudServer);
-            pItem = new QListWidgetItem();
-            // Can't use setupMudProfile(...) here as we do not set the icon in the same way:
-            setItemName(pItem, mudServer);
-
+    // This badly named method actually creates icons for all the
+    // profiles found - and adds them to the nameToItemMap passed (by reference):
+    setProfileIcon(nameToItemMap);
+
+    // Now read the stored order - and the size we stored for this dialog:
+    QSize desiredDialogSize{};
+    QStringList existingProfilesOrderedList = loadProfilesOrderAndDlgSize(&desiredDialogSize);
+    // First make a pass through the QMap of QListWidgetItems - and identify
+    // those whose name is NOT in the stored, ordered, list of profiles, so we
+    // can place them first:
+    QMutableMapIterator<QString, QListWidgetItem*> itProfile(nameToItemMap);
+    while (itProfile.hasNext()) {
+        itProfile.next();
+        if (!existingProfilesOrderedList.contains(itProfile.key(), Qt::CaseInsensitive)) {
+            // This is a new one that we don't have an order position for, so
+            // add it straight into the listWiget so it will be above the
+            // "known" ones - and give it a green background to highlight that
+            // it is a new one:
+            auto pItem = itProfile.value();
+            // Remove it from the QMap so there are less to deal with - so the
+            // second pass is simpler:
+            itProfile.remove();
+            // Insert it into the QListWidget
             profiles_tree_widget->addItem(pItem);
-            description = getDescription(qsl("mudlet.org"));
-            if (!description.isEmpty()) {
-                pItem->setToolTip(utils::richText(description));
-            }
+            // Now that it is owned by the QListWidget the setHidden(bool)
+            // method will operate to hide it if requested:
+            pItem->setHidden(pItem->data(csmHiddenRole).toBool());
```

Comment:
I found this out the hard way - it was breaking my prototype code as I was calling `QListWidgetItem::setHidden(true)` but that was lost/didn't stick if the `QListWidgetItem` is **not** in a `QListWidget`!

:rage: